components/datetime/src/lib.rs

```diff
@@ -146,18 +146,18 @@ impl<'d> DateTimeFormat<'d> {
     /// use icu_datetime::date::MockDateTime;
     /// use icu_provider::inv::InvariantDataProvider;
     ///
-    /// let lid = langid!("en");
+    /// let locale = langid!("en").into();
     ///
     /// let provider = InvariantDataProvider;
     ///
     /// let options = DateTimeFormatOptions::default();
     ///
-    /// let dtf = DateTimeFormat::try_new(lid, &provider, &options);
+    /// let dtf = DateTimeFormat::try_new(locale, &provider, &options);
     ///
     /// assert_eq!(dtf.is_ok(), true);
     /// ```
     pub fn try_new<D: DataProvider<'d, provider::gregory::DatesV1> + ?Sized>(
-        langid: LanguageIdentifier,
+        locale: Locale,
```

Comment:
So, up to you, you can change it already to `Into<Locale>` as per https://github.com/unicode-org/icu4x/blob/master/components/locid/src/locale.rs#L147

Reply:
I'm going to merge this as-is, since it will be revisited anyway in #447.